### ISSUE TITLE
Can't connect from Linux Mint (works from iOS using OpenVPN Connect)

### ISSUE THREAD

**Issue by mmeh** [pivpn/pivpn#120]
Hi. I have used the PiVPN script to successfully set up a VPN on a Raspberry Pi 3 running Raspbian Jessie Lite. It was pretty nice and straightforward.

Using my iPhone, I have imported the `username.ovpn` for my one and only client into the OpenVPN Connect app, and connection works like a charm.

However, on my laptop I am running Linux Mint 17.3 (Cinnamon), and I have attempted to set up the connection. I have copied over the `username.ovpn`, created a folder `~/.ovpn` and I have used the OVPN Splitter script (https://gist.github.com/penguinpowernz/36883ba5c8bf51bdb5d1) to extract the `ca.crt`, `client.crt`, `client.key` and `ta.key` into my `~/.ovpn` directory, and also have a stripped down `username.ovpn` file which links those files. I now used that new `username.ovpn` file and imported it into the Network Manager, and it set up the IP and certificate/key files as expected. However, when I try to start the connection, it seems to work for some time and then gives up.

I have taken the following extract from `/var/log/syslog`, but I don't see what the problem is:
```
NetworkManager[947]: <info> Starting VPN service 'openvpn'...
NetworkManager[947]: <info> VPN service 'openvpn' started (org.freedesktop.NetworkManager.openvpn), PID 5490
NetworkManager[947]: <info> VPN service 'openvpn' appeared; activating connections
NetworkManager[947]: <info> VPN plugin state changed: starting (3)
NetworkManager[947]: <info> VPN connection 'username' (Connect) reply received.
nm-openvpn[5498]: OpenVPN 2.3.2 x86_64-pc-linux-gnu [SSL (OpenSSL)] [LZO] [EPOLL] [PKCS11] [eurephia] [MH] [IPv6] built on Dec  1 2014
nm-openvpn[5498]: NOTE: the current --script-security setting may allow this configuration to call user-defined scripts
nm-openvpn[5498]: WARNING: file '/home/username/.ovpn/client.key' is group or others accessible
nm-openvpn[5498]: WARNING: file '/home/username/.ovpn/ta.key' is group or others accessible
nm-openvpn[5498]: Control Channel Authentication: using '/home/username/.ovpn/ta.key' as a OpenVPN static key file
nm-openvpn[5498]: UDPv4 link local: [undef]
nm-openvpn[5498]: UDPv4 link remote: [AF_INET]xxx.xxx.xxx.xxx:1194
NetworkManager[947]: <warn> VPN connection 'username' (IP Config Get) timeout exceeded.
NetworkManager[947]: <info> Policy set 'Auto oppenheimer' (wlan0) as default for IPv4 routing and DNS.
nm-openvpn[5498]: SIGTERM[hard,] received, process exiting
NetworkManager[947]: <info> VPN service 'openvpn' disappeared
```
Any help is much appreciated!

**Comment by redfast00** [comment]
Are ypu using the same .ovpn file?  You should generate a new one for each client.

**Comment by mmeh** [comment]
I am using the same `.ovpn` file, yes, but that shouldn't make a different anyway, right? I basically wanted to use it on my laptop, but since it failed I tried it on the iPhone after, where it works fine.

**Comment by redfast00** [comment]
Have you tried using the commandline version?

**Comment by mmeh** [comment]
I never really tried using OpenVPN from the command line. However, when I go to `~/.ovpn` where my configs are stored, and I do `openvpn --config username.ovpn` I get the following error:

`Options error: Unrecognized option or missing parameter(s) in username.ovpn:11: tls-version-min (2.3.2)`
`Use --help for more information.`


If I remove line `tls-version-min 1.2` from the `username.ovpn` and try again, this is what happens:

`~/.ovpn $ openvpn --config username.ovpn `
`Tue Oct 18 10:35:22 2016 OpenVPN 2.3.2 x86_64-pc-linux-gnu [SSL (OpenSSL)] [LZO] [EPOLL] [PKCS11] [eurephia] [MH] [IPv6] built on Dec  1 2014`
`Enter Private Key Password:`
`Tue Oct 18 10:35:28 2016 WARNING: this configuration may cache passwords in memory -- use the auth-nocache option to prevent this`
`Tue Oct 18 10:35:28 2016 WARNING: file 'client.key' is group or others accessible`
`Tue Oct 18 10:35:28 2016 WARNING: file 'ta.key' is group or others accessible`
`Tue Oct 18 10:35:28 2016 Control Channel Authentication: using 'ta.key' as a OpenVPN static key file`
`Tue Oct 18 10:35:28 2016 UDPv4 link local: [undef]`
`Tue Oct 18 10:35:28 2016 UDPv4 link remote: [AF_INET]xxx.xxx.xxx.xxx:1194`
`Tue Oct 18 10:36:28 2016 TLS Error: TLS key negotiation failed to occur within 60 seconds (check your network connectivity)`
`Tue Oct 18 10:36:28 2016 TLS Error: TLS handshake failed`
`Tue Oct 18 10:36:28 2016 SIGUSR1[soft,tls-error] received, process restarting`
`Tue Oct 18 10:36:30 2016 UDPv4 link local: [undef]`
`Tue Oct 18 10:36:30 2016 UDPv4 link remote: [AF_INET]xxx.xxx.xxx.xxx:1194`
`Tue Oct 18 10:37:30 2016 TLS Error: TLS key negotiation failed to occur within 60 seconds (check your network connectivity)`
`Tue Oct 18 10:37:30 2016 TLS Error: TLS handshake failed`
`Tue Oct 18 10:37:30 2016 SIGUSR1[soft,tls-error] received, process restarting`
`Tue Oct 18 10:37:32 2016 UDPv4 link local: [undef]`
`Tue Oct 18 10:37:32 2016 UDPv4 link remote: [AF_INET]xxx.xxx.xxx.xxx:1194`
`^CTue Oct 18 10:37:36 2016 event_wait : Interrupted system call (code=4)`
`Tue Oct 18 10:37:36 2016 SIGINT[hard,] received, process exiting`

So, the TLS handshake seems to be failing. I guess the PiVPN requires TLS 1.2, and my OpenVPN client seems to have an issue with that.

**Comment by mmeh** [comment]
After building OpenVPN 2.3.12 on my Linux machine, I tried doing `sudo openvpn username.ovpn` and it seems to work. However, there are still some errors. I have pasted the output here: https://nopaste.me/view/902ff889#6xvmcWRUMLNjurr0JYeIVEGvkKqGh7B3

What's more is that it only works via the terminal. If I use the Network manager, I get the same output as previously in `/var/log/syslog`.

**Comment by redfast00** [comment]
Hi, could you give me your `.ovpn` file? Please comment out your IP address/hostname and the keys.

**Comment by mmeh** [comment]
Hi redfast00, sure, this is the `.ovpn` that I copied from the PiVPN server: https://nopaste.me/view/408db40a#tRrtZgQ0ephCwepZoc9M7hvG8w3LqAnQ

**Comment by redfast00** [comment]
Okay, looks good. How about `sudo cat /etc/openvpn/server.conf` (server side) and the output of `sudo openvpn --version` (laptop)? I'm slowly running out of ideas of what it could be...

**Comment by mmeh** [comment]
Sure, this is the `server.conf` file: https://nopaste.me/view/eb09b1a7#3QbNQTO1yUldLs9CMArPkNGm5Tjz8tr4

This is `sudo openvpn --version`:
https://nopaste.me/view/96d59c22#53OkT7jjYZu8RUdrIhqiV6tQmLyp3EQu

**Comment by mmeh** [comment]
Any ideas?

**Comment by redfast00** [comment]
I don't really know at this point. It might be because NetworkManager uses the old version of OpenVPN?

**Comment by brandonlmain** [comment]
Did you port forward 1194 on your router to the pis static ip address? I was having handshake errors before doing so.

Also I see your OpenVPN version is 2.3.12 and this looks like it's been an issue with using tls-version-min 1.2 on OpenVPN client since 2.3.4:
[https://community.openvpn.net/openvpn/ticket/401#no1](url)

**Comment by redfast00** [comment]
@brandonlmain The portforward probably isn't the issue, since it works just fine with an iPhone. It might be the version though.
@mmeh: I looked at the paste again. The output doesn't differ much from mine on Android/Debian. If it works via the commandline (you can verify it actually works by doing a packet capture with Wireshark on `wlan0`) and it doesn't in the NetworkManager GUI, it probably means NetworkManager still uses the old version.

**Comment by mmeh** [comment]
@brandonlmain Yes, the port is forwarded, and as @redfast00 says it works fine in two cases.

@redfast00 I will try if I can remove the old version of OpenVPN or otherwise change how NetworkManager picks the OpenVPN.

**Comment by mmeh** [comment]
@redfast00 It seems I managed to fix it by first removing OpenVPN 2.3.12, and then installing it again  using the `checkinstall` program which appropriately registers it with Synaptic. Now it is working nicely. Thanks a lot!
